### DISCUSSION TITLE
sentinel mode， lost "+switch-master"  message, should client add watchdog thread

### DISCUSSION THREAD

**Discussion by chenshi5012** [xetorthio/jedis#3525] [Ideas]
I get some client and master-server in "vpc1"

![image](https://github.com/redis/jedis/assets/10794822/89141be3-dd9d-49df-8e63-8e9bab42c2da)

redis-slave in "vpc2",

some sence,  cause the network issue,  failover happened,   after some time  the network issue resovle. 

but the client lost the  channel message  from  "+switch-master" .  

happened some time under test envoirment.  

![image](https://github.com/redis/jedis/assets/10794822/38ace2a0-c6ee-4c41-8a4d-d9b43056cf0f)

can we make some cheduled active detection. to  find out the master-failover.  instead of the subscribe message  form "+switch-master" 。

**Comment by sazzad16** [comment]
PRs are welcome.
Please allow configuration; i.e. enable/disable, schedule time, etc.
